File: sorts/insert_interval.rb
def insert(intervals, new_interval)
    result = []

    intervals.each_with_index do |interval, i|

        # if interval after
        if new_interval[1] < interval[0]
            
            result << new_interval
            return result + intervals[i...intervals.size]
        # if interval before
        elsif new_interval[0] > interval[1]
            result << interval
        # overlapping
        else
            new_interval = [[interval[0],new_interval[0]].min, [interval[1], new_interval[1]].max]
        end
    end
    result << new_interval
    result

end

File: sorts/insert_interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        result = []

        for i, interval in enumerate(intervals):
            # if interval after
            if newInterval[1] < interval[0]:
                
                result.append(newInterval)
                return result + intervals[i:]
            # if interval before
            elif newInterval[0] > interval[1]:
                result.append(interval)
            # overlapping
            else:
                newInterval = [min(interval[0], newInterval[0]), max(interval[1], newInterval[1])]
            
        result.append(newInterval)
        return result